packages/sync-repo-settings/src/config.ts

```diff
@@ -0,0 +1,122 @@
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import {RepoConfig} from './types';
+import {Octokit} from '@octokit/rest';
+import {logger} from 'gcf-utils';
+import Ajv from 'ajv';
+import yaml from 'js-yaml';
+import schema from './schema.json';
+
+// configure the schema validator once
+const ajv = new Ajv();
+const validateSchema = ajv.compile(schema);
+
+export const configFileName = 'sync-repo-settings.yaml';
+
+export interface GetConfigOptions {
+  octokit: Octokit;
+  owner: string;
+  repo: string;
+}
+
+export interface ValidateConfigResponse {
+  isValid: boolean;
+  errorText?: string;
+  config?: RepoConfig;
+}
+
+/**
+ * Given a config in its raw yaml form, validate that it matches our config
+ * schema.  Return any validation errors from ajv.
+ * @param configYaml Raw text containing the YAML to validate.
+ * @returns
+ */
+export async function validateConfig(configYaml: string) {
+  const config = yaml.load(configYaml) as RepoConfig;
+  let isValid = false;
+  let errorText: string | undefined;
+  if (typeof config === 'object') {
+    isValid = await validateSchema(config);
+    if (!isValid) {
+      errorText = JSON.stringify(validateSchema.errors, null, 4);
+    }
+  } else {
+    errorText = `${configFileName} is not valid YAML 😱`;
+  }
+  return {isValid, errorText};
+}
+
+/**
+ * Allow repositories to optionally provide their own, localized config.
+ * Check the `.github/sync-repo-settings.yaml` file, and if available,
+ * use that config over any config broadly provided here.
+ */
+export async function getConfig(
+  options: GetConfigOptions
+): Promise<RepoConfig | null> {
+  let config!: RepoConfig | null;
+
+  // attempt to fetch config for the local repository
+  try {
+    config = await getConfigFile(options);
+    return config;
+  } catch (err) {
+    if (err.status === 404) {
+      logger.info('No local config found.');
+    } else {
+      err.message = `Error reading configuration: ${err.message}`;
+      logger.error(err);
+    }
+  }
+
+  // attempt to fetch config for the org level repo
+  try {
+    config = await getConfigFile({
+      octokit: options.octokit,
+      owner: options.owner,
+      repo: '.github',
+    });
+    return config;
+  } catch (err) {
+    if (err.status === 404) {
+      logger.info('No remote config found.');
+    } else {
+      err.message = `Error reading configuration: ${err.message}`;
+      logger.error(err);
+    }
+  }
+  return null;
+}
+
+/**
+ * For a given repository, fetch the config yaml.
+ * @param options GetConfigOptions
+ * @returns an individual Repo config
+ */
+async function getConfigFile(
```

Comment:
I wonder if it would be worth pulling config handling logic up to `gcf-utils`, so that we can use it for other repos?

Reply:
Not blocker for this PR.

Reply:
For other bots, for now, I'm content with the default behavior where the `.github` repository level config flat overrides the per-repo config.  This bot is "special" in the way I am trying to treat it for sure.  If the pattern plays out and we want to use it elsewhere, I'd happily up-level it to the util library 👍